File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/datatype/FloatType.kt
package dynabuffers.ast.datatype

import dynabuffers.api.IRegistry
import dynabuffers.api.ISerializable
import dynabuffers.api.IType
import java.nio.ByteBuffer

class FloatType : IType, ISerializable {

    override fun size(value: Any, registry: IRegistry) = 4

    override fun serialize(value: Any, buffer: ByteBuffer, registry: IRegistry) {
        buffer.putFloat(float(value))
    }

    override fun deserialize(buffer: ByteBuffer, registry: IRegistry) = buffer.float

    private fun float(obj: Any) = obj.toString().toFloat()

}

File: dynabuffers-python/dynabuffers/ast/datatype/FloatType.py
from dynabuffers.api.ISerializable import ISerializable


class FloatType(ISerializable):

    def size(self, value, registry):
        return 4

    def serialize(self, value, buffer, registry):
        buffer.putFloat(value if isinstance(value, float) else float(value))

    def deserialize(self, buffer, registry):
        return buffer.getFloat()
